### ISSUE TITLE
New Cvxportfolio versions will have the General Public License

### ISSUE THREAD

**Issue by enzbus** [cvxgrp/cvxportfolio#166]
## Tl;DR

Starting from Cvxportfolio `1.4`, the software license will be the **General Public License version 3**. 

## Retro-compatibility

The current license, `Apache 2.0`, poses no restrictions on this change, nor it requires written agreement from contributors. [Source 1](https://www.gnu.org/licenses/license-list.html#apache2), [source 2](https://www.apache.org/licenses/GPL-compatibility.html): both the Free Software Foundation and Apache Foundation agree that `Apache 2.0` licensed software can be included in `GPLv3` projects, no questions asked.

Cvxportfolio versions up to `1.3` continue of course to be licensed as they currently are. I will continue to maintain this legacy version, with security fixes, but my main focus will be on maintaining and developing Cvxportfolio `1.4` and further.

## For users: what changes

In most cases, not much. You can continue using Cvxportfolio in your personal projects, for fun or for profit. If your company uses Cvxportfolio you can also continue to use it. If you build software that uses Cvxportfolio as a dependency, and you only use it yourself or within your company, you can continue to do so. **If you re-distribute software that contains any part, modified or unmodified, of Cvxportfolio, or calls it in any way**, such software has to be licensed under the `GPL`, or a compatible license. If you don't re-distribute it, meaning you use it internally in your organization or on your computer, no such restrictions apply. *This is not legal advice*, but to the best of my knowledge that's what the General Public License means in this context. Do consult a lawyer if you are in doubt.

## For contributors: what changes

Your contributions are now protected by the `GPL`. [Read about it here](https://www.gnu.org/licenses/gpl-3.0.html). In short, it gives you the assurance that nobody is going to "steal" your intellectual property, and your contributions will remain public.

## For companies: what changes

Read the section above about users. If your company uses Cvxportfolio in their internal software repositories, you can continue to do so. If you are a hedge fund, family office, sovereign wealth fund, ..., you can continue using Cvxportfolio for your portfolio optimization and back-testing workflows, and you're welcome to do so. If you are a **software company** and plan to **distribute** software, which can be a web application, that contains any dependency on modified or unmodified Cvxportfolio code, such software will have to be released under the `GPL` or compatible license, and its source will have to be public. Note that this doesn't prevent you from making money from it! For example, you can sell support about it, sell access to hosted instances, and so on.

## Repository

~~The new versions will be in the `main` branch of the repository. The branch shown on the homepage will remain `master` for now, and `master` will continue to host the legacy, pre-`GPL` versions. Clear notes will be shown.~~
